java-checks/src/main/java/org/sonar/java/checks/AnonymousClassShouldBeLambdaCheck.java

```diff
@@ -76,14 +79,38 @@ public void visitNewClass(NewClassTree tree) {
   }
 
   private static boolean isSAM(ClassTree classBody) {
-    if(hasOnlyOneMethod(classBody.members()) && classBody.symbol().isTypeSymbol()) {
-      // Verify class body is a subtype of an interface
+    if (hasOnlyOneMethod(classBody.members())) {
       JavaSymbol.TypeJavaSymbol symbol = (JavaSymbol.TypeJavaSymbol) classBody.symbol();
-      return symbol.getInterfaces().size() == 1 && symbol.getSuperclass().is("java.lang.Object");
+      List<JavaType> interfaces = symbol.getInterfaces();
+      // should be anonymous class of interface and not abstract class
+      return interfaces.size() == 1
+        && symbol.getSuperclass().is("java.lang.Object")
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define a constant instead of duplicating this literal "java.lang.Object" 3 times. ([squid:S1192](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.java%3Ajava&pullRequest=1946&issues=AWJN6OzXs_VgbIaGzJ0p&open=AWJN6OzXs_VgbIaGzJ0p)

[ISSUE_KEY]: AWJN6OzXs_VgbIaGzJ0p